project_classification/project_classification.py

```diff
@@ -0,0 +1,77 @@
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import api, models, fields
+
+
+class ProjectClassification(models.Model):
+    _name = "project.classification"
+    _description = "Project classification"
+
+    name = fields.Char('Classification Name', required=True, size=64)
+    project_id = fields.Many2one('account.analytic.account',
+                                 'Parent project',
+                                 help="The parent project that will be set "
+                                 "when choosing this classification in "
+                                 "a project.",
+                                 required=True)
+    to_invoice = fields.Many2one('hr_timesheet_invoice.factor',
+                                 'Reinvoice Costs',
+                                 help="Fill this field if you plan to "
+                                 "automatically generate invoices based "
+                                 "on the costs in this classification")
+    currency_id = fields.Many2one('res.currency', 'Currency')
+    user_id = fields.Many2one('res.users', 'Account Manager')
+    pricelist_id = fields.Many2one('product.pricelist',
+                                   'Sale Pricelist')
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    classification_id = fields.Many2one('project.classification',
+                                        'Classification',
+                                        help="This will automatically set "
+                                        "the parent project as well as other "
+                                        "default values defined for this kind "
+                                        "of project (like pricelist, "
+                                        "invoice factor,..)",
+                                        required=False)
+
+    child_project_complete_ids = fields.Many2many(
+        'project.project',
+        compute='_child_project_compute',
+        string="Project Hierarchy")
+
+    @api.depends('child_complete_ids.project_ids')
+    def _child_project_compute(self):
+        for project in self:
+            child_projects = project.mapped('child_complete_ids.project_ids')
+            project.child_project_complete_ids = child_projects
+
+    def onchange_classification_id(self, cr, uid, ids, classification_id):
```

Comment:
Any reason to keep it in old API ? Would be better with new API as it would make it simple to inherit